third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -44,6 +44,50 @@ int getWmmaVersion(StringRef archGen) {
   return 0;
 }
 
+// Check if the result of current tl.dot is used as the operand(0)
+// of another tl.dot
+bool isChainDotHead(tt::DotOp &dotOp) {
+  auto filter = [&dotOp](Operation *op) {
```

Comment:
Typically we want to have self-documenting function/variable name for readability. It's not a big concern here given it's a oneliner, but `isInSameRegion` would be better than a general `filter` name here.